Compilation failes for DEAL_II_WITH_64BIT_INDICES="ON"

The master branch does not compile after the recent changes for 64 bit indices. E.g.: 
```bash
/home/munch/sw-index/dealii/source/dofs/dof_renumbering.cc: In function ‘void dealii::DoFRenumbering::compute_Cuthill_McKee(std::vector<long unsigned int>&, const DoFHandlerType&, bool, bool, const std::vector<long unsigned int>&)’:
/home/munch/sw-index/dealii/source/dofs/dof_renumbering.cc:433:67: error: no matching function for call to ‘reverse_permutation(std::vector<long unsigned int>&)’
  433 |           new_indices = Utilities::reverse_permutation(new_indices);
      |                                                                   ^
In file included from /home/munch/sw-index/dealii/include/deal.II/base/tensor.h:27,
                 from /home/munch/sw-index/dealii/include/deal.II/base/point.h:23,
                 from /home/munch/sw-index/dealii/include/deal.II/base/quadrature.h:22,
                 from /home/munch/sw-index/dealii/include/deal.II/base/quadrature_lib.h:22,
                 from /home/munch/sw-index/dealii/source/dofs/dof_renumbering.cc:16:
/home/munch/sw-index/dealii/include/deal.II/base/utilities.h:566:3: note: candidate: ‘std::vector<unsigned int> dealii::Utilities::reverse_permutation(const std::vector<unsigned int>&)’
  566 |   reverse_permutation(const std::vector<unsigned int> &permutation);
      |   ^~~~~~~~~~~~~~~~~~~
/home/munch/sw-index/dealii/include/deal.II/base/utilities.h:566:56: note:   no known conversion for argument 1 from ‘std::vector<long unsigned int>’ to ‘const std::vector<unsigned int>&’
  566 |   reverse_permutation(const std::vector<unsigned int> &permutation);
      |                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~
/home/munch/sw-index/dealii/include/deal.II/base/utilities.h:582:3: note: candidate: ‘std::vector<long long unsigned int> dealii::Utilities::reverse_permutation(const std::vector<long long unsigned int>&)’
  582 |   reverse_permutation(const std::vector<unsigned long long int> &permutation);
      |   ^~~~~~~~~~~~~~~~~~~
/home/munch/sw-index/dealii/include/deal.II/base/utilities.h:582:66: note:   no known conversion for argument 1 from ‘std::vector<long unsigned int>’ to ‘const std::vector<long long unsigned int>&’
  582 |   reverse_permutation(const std::vector<unsigned long long int> &permutation);

```

Also there are some warnings...

I have a solution for the compile error. I don't like the warning, though:
```
In file included from /home/kronbichler/sw/dealii_git/source/lac/vector.cc:16,
                 from /home/kronbichler/sw/build_deal_64bit/source/lac/unity_2.cc:9:
/home/kronbichler/sw/dealii_git/include/deal.II/lac/vector.templates.h: In instantiation of ‘void dealii::Vector<Number>::block_write(std::ostream&) const [with Number = double; std::ostream = std::basic_ostream<char>]’:
/home/kronbichler/sw/build_deal_64bit/source/lac/vector.inst:10:16:   required from here
/home/kronbichler/sw/dealii_git/include/deal.II/lac/vector.templates.h:931:25: warning: format ‘%llu’ expects argument of type ‘long long unsigned int’, but argument 3 has type ‘dealii::Vector<double>::size_type’ {aka ‘long unsigned int’} [-Wformat=]
  931 |   std::sprintf(buf, "%llu", sz);
      |                      ~~~^   ~~
      |                         |   |
      |                         |   dealii::Vector<double>::size_type {aka long unsigned int}
      |                         long long unsigned int
      |                      %lu
```
We could of course switch to `%lu` as the compiler suggests. But that is wrong on platforms where `unsigned long int` is not a 64 bit integer, e.g. the 32-bit x86 architecture. Should we convert the number to `unsigned long long int` before printing?

--- a/include/deal.II/lac/la_vector.templates.h
+++ b/include/deal.II/lac/la_vector.templates.h
@@ -592,15 +592,14 @@ namespace LinearAlgebra
 
     // Other version of the following
     //  out << size() << std::endl << '[';
-    // Reason: operator<< seems to use some resources  that lead to problems in
-    // a multithreaded environment.
-    const size_type sz = this->size();
-    char            buf[16];
-#ifdef DEAL_II_WITH_64BIT_INDICES
+    // Reason: operator<< seems to use some resources that lead to problems in
+    // a multithreaded environment.  We convert the size index to
+    // unsigned long long int that is at least 64 bits to be able to output it
+    // on all platforms, since std::uint64_t is not in C.
+    const unsigned long long int sz = this->size();
+    char                         buf[16];
+
     std::sprintf(buf, "%llu", sz);
-#else
-    std::sprintf(buf, "%u", sz);
-#endif
     std::strcat(buf, "\n[");
 
     out.write(buf, std::strlen(buf));

--- a/include/deal.II/lac/vector.templates.h
+++ b/include/deal.II/lac/vector.templates.h
@@ -920,18 +920,15 @@ Vector<Number>::block_write(std::ostream &out) const
 
   // other version of the following
   //  out << size() << std::endl << '[';
-  // reason: operator<< seems to use
-  // some resources that lead to
-  // problems in a multithreaded
-  // environment
-  const size_type sz = size();
-  char            buf[16];
 
-#ifdef DEAL_II_WITH_64BIT_INDICES
+  // reason: operator<< seems to use some resources that lead to problems in a
+  // multithreaded environment. We convert the size index to
+  // unsigned long long int that is at least 64 bits to be able to output it on
+  // all platforms, since std::uint64_t is not in C.
+  const unsigned long long int sz = size();
+  char                         buf[16];
+
   std::sprintf(buf, "%llu", sz);
-#else
-  std::sprintf(buf, "%u", sz);
-#endif
   std::strcat(buf, "\n[");
 
   out.write(buf, std::strlen(buf));

